Source/Core/Core/HW/GCMemcard/GCMemcard.cpp

```diff
@@ -1389,6 +1387,47 @@ s32 GCMemcard::PSO_MakeSaveGameValid(const Header& cardheader, const DEntry& dir
   return 1;
 }
 
+Header::Header(int slot, u16 size_mbits, bool shift_jis)
+{
+  // Nintendo format algorithm.
+  // Constants are fixed by the GC SDK
+  // Changing the constants will break memory card support
+  memset(this, 0xFF, BLOCK_SIZE);
+  m_size_mb = size_mbits;
+  m_encoding = shift_jis ? 1 : 0;
+  u64 rand = Common::Timer::GetLocalTimeSinceJan1970() - ExpansionInterface::CEXIIPL::GC_EPOCH;
+  m_format_time = rand;
+  for (int i = 0; i < 12; i++)
+  {
+    rand = (((rand * (u64)0x0000000041c64e6dULL) + (u64)0x0000000000003039ULL) >> 16);
+    m_serial[i] = (u8)(g_SRAM.settings_ex.flash_id[slot][i] + (u32)rand);
+    rand = (((rand * (u64)0x0000000041c64e6dULL) + (u64)0x0000000000003039ULL) >> 16);
+    rand &= (u64)0x0000000000007fffULL;
+  }
+  m_sram_bias = g_SRAM.settings.rtc_bias;
+  m_sram_language = static_cast<u32>(g_SRAM.settings.language);
+  // TODO: determine the purpose of m_unknown_2
+  // 1 works for slot A, 0 works for both slot A and slot B
+  memset(m_unknown_2.data(), 0,
+         m_unknown_2.size());  // = _viReg[55];  static vu16* const _viReg = (u16*)0xCC002000;
```

Comment:
This comment has just been moved along, but do we know what it _means_?

Reply:
Not sure? I guess it's meant to imply that it copies the value from the `VI_DTV_STATUS` register, but I don't think that makes a lot of sense.